lib/access-context.js

```diff
@@ -126,8 +128,20 @@ AccessContext.prototype.addPrincipal = function(principalType, principalId, prin
 AccessContext.prototype.getUserId = function() {
   for (var i = 0; i < this.principals.length; i++) {
     var p = this.principals[i];
+    // the principalType must either be 'USER' or permit to resolve a valid user model
     if (p.type === Principal.USER) {
       return p.id;
+    } else {
+      if (!this.registry) return null;
+      var model = this.registry.findModel(p.type);
+      if (!model) return null;
+      var modelBase = model.definition.settings.base;
+      var isUserBasedModel = model.modelName === 'User' ||
+         (typeof(modelBase) === 'string' && modelBase === 'User') ||
+         modelBase.modelName === 'User';
```

Comment:
you're right, i noted that too. we should test recursively up to the top User model. I'll change the implem and add a test

**quick question:** 
i'm looking for a clean way to get the original user base model name by code rather than hard coding it `'User'` in the conditional tests? Are you ok with requiring the model definition to do this?
``` javascript
// in file header, parsed at app boot
var builtInUserModel = require('../common/models/user.json');

// in method:
var builtInUserModelName = builtInUserModel.name;
```